packages/plugin-ext/src/main/browser/view/tree-views-main.tsx

```diff
@@ -271,6 +359,30 @@ export class TreeViewWidget extends TreeWidget {
         }
     }
 
+    getIconPath(icon: string | { light: string; dark: string }): string {
+        let iconPath: string;
```

Comment:
Can we handle it in a similar way how we handle command icons: https://github.com/theia-ide/theia/blob/e710223dd3ee8acb70569a3cf8b5266574f47386/packages/plugin-ext/src/main/browser/plugin-contribution-handler.ts#L160-L173

Reply:
I am not a css expert, but again, this url is provided by an interface implemented by the extension and comes late after css is loaded (can be when you expand a tree folder for example.

Reply:
It is the same for commands. This code dynamically inserts new css rules and updates it based on theme changes.

Reply:
I will try to convert it this way, because as you said, the advantage is that it is sensitive to theme change.
but a question: 
Since tree-nodes are dynamic (and potentially each has different icon) this may produce many rules - is that an issue?

Reply:
Ok I did. see latest commit.